ethereum_history_api/circuits/lib/src/rlp/decode.nr

```diff
@@ -49,49 +54,30 @@ pub fn extract_payload_len<N>(arr: [u8; N], lenlen: u64) -> u64 {
 pub fn decode_to_rlp_header<N>(arr: [u8; N]) -> RlpHeader {
     let prefix = arr[0];
 
-    // Prefix range indicators
-    let is_single_byte = (prefix < 0x80); // Single byte
-    let is_short_string = (prefix < 0xb8); // 0-55 byte string
-    let is_long_string = (prefix < 0xc0); // >55-byte string
-    let is_short_list = (prefix < 0xf8); // 0-55 byte list
-    // Else >55-byte list
-
-    let offset = strict_if(
-        is_single_byte,
-        0,
-        strict_if(
-            is_short_string,
-            1,
-            strict_if(
-                is_long_string,
-                wrapping_sub(1 + prefix, 0xb7) as u64,
-                strict_if(is_short_list, 1, wrapping_sub(1 + prefix, 0xf7) as u64)
-            )
-        )
-    );
-
-    // To compute length, first determine how many of the following bytes contain the length of the payload.
-    let lenlen = is_long_string as u64 * (1 - is_short_string as u64) * wrapping_sub(prefix, 0xb7) as u64
-        + (1 - is_short_list as u64) * wrapping_sub(prefix, 0xf7) as u64;
-
-    // Set length according to whether it is contained in the prefix (i.e. lenlen == 0).
-    let len = strict_if(
-        lenlen == 0,
-        strict_if(
-            is_single_byte,
-            1,
-            strict_if(
-                is_short_string,
-                wrapping_sub(prefix, 0x80) as u64,
-                wrapping_sub(prefix, 0xc0) as u64
-            )
-        ),
-        extract_payload_len(arr, lenlen)
-    );
-
-    let data_type = (prefix >= 0xc0) as u64;
-
-    RlpHeader { offset, length: len, data_type }
+    let mut data_type = STRING;
+    if (prefix >= 0xc0) {
+        data_type = LIST;
+    }
+
+    if (prefix < 0x80) { // single byte
+        RlpHeader { offset: 0, length: 1, data_type }
+    } else if (prefix < 0xb8) { // 0-55 byte string
+        RlpHeader { offset: 1, length: wrapping_sub(prefix, 0x80) as u64, data_type }
```

Comment:
I wonder if we need `as u64` or type inference takes care of it?

Reply:
We need it